From: Christian Lamparter <chunkeey@gmail.com>
Subject: [PATCH] ath79: port cybertan_part from ar71xx

This patch ports the cybertan_part code from ar71xx and converts the
driver to a DT-supported mtd parser. As a result, it will no longer
add the u-boot, nvram and art partitions, which were never part of
the special Cybertan header.

Instead these partitions have to be specified in the DT, which has the
upside of making it possible to add properties (i.e.: read-only), labels
and references to these important partitions.

Submitted-by: Christian Lamparter <chunkeey@gmail.com>
---
 drivers/mtd/parsers/Makefile                  | 1 +
 drivers/mtd/parsers/Kconfig                   | 8 ++++++++
 2 files changed, 9 insertions(+)

--- a/drivers/mtd/parsers/Makefile
+++ b/drivers/mtd/parsers/Makefile
@@ -8,6 +8,7 @@ obj-$(CONFIG_MTD_OF_PARTS)		+= ofpart.o
 ofpart-y				+= ofpart_core.o
 ofpart-$(CONFIG_MTD_OF_PARTS_BCM4908)	+= ofpart_bcm4908.o
 ofpart-$(CONFIG_MTD_OF_PARTS_LINKSYS_NS)+= ofpart_linksys_ns.o
+obj-$(CONFIG_MTD_PARSER_CYBERTAN)	+= parser_cybertan.o
 obj-$(CONFIG_MTD_PARSER_IMAGETAG)	+= parser_imagetag.o
 obj-$(CONFIG_MTD_AFS_PARTS)		+= afs.o
 obj-$(CONFIG_MTD_PARSER_TPLINK_SAFELOADER)	+= tplink_safeloader.o
--- a/drivers/mtd/parsers/Kconfig
+++ b/drivers/mtd/parsers/Kconfig
@@ -107,6 +107,14 @@ config MTD_OF_PARTS_LINKSYS_NS
 	  two "firmware" partitions. Currently used firmware has to be detected
 	  using CFE environment variable.
 
+config MTD_PARSER_CYBERTAN
+	tristate "Parser for Cybertan format partitions"
+	depends on MTD && (ATH79 || COMPILE_TEST)
+	help
+	  Cybertan has a proprietory header than encompasses a Broadcom trx
+	  header. This driver will parse the header and take care of the
+	  special offsets that result in the extra headers.
+
 config MTD_PARSER_IMAGETAG
 	tristate "Parser for BCM963XX Image Tag format partitions"
 	depends on BCM63XX || BMIPS_GENERIC || COMPILE_TEST
